### PR TITLE
B2B-869 - Add concepts

### PR DESCRIPTION
**Pull request by Dominik28111**

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -40,6 +40,10 @@
   * [Professional Edition](products/editions/professional-edition.md)
   * [Enterprise Edition](products/editions/enterprise-edition/README.md)
     * [B2B Suite](products/editions/enterprise-edition/b2b-suite.md)
+    * [Concepts](products/editions/enterprise-edition/concepts/README.md)
```
Comment (bschulzebaek) [inline SUMMARY.md:43]: These pages should be located inside the "B2B Suite" tree, not next to it.

File: products/editions/enterprise-edition/concepts/basic-conventions.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+| Attributes                                                                                         | All attribute names start with `swag_b2b_`                                          |
+| Subscriber                                                                                         | All subscriber methods are named in accordance to their function, not to the event. |
+| Tests                                                                                              | All test methods are in snake case                                                  |
+| All test methods start with `test_`                                                                |                                                                                     |
+| Templates                                                                                          | All new layout modules are wrapped in `b2b--*` class containers                     |
+| modules reuse the template style of shopware                                                       |                                                                                     |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/concepts/basic-conventions.md:18]: Capitalize "modules"

File: products/editions/enterprise-edition/concepts/basic-conventions.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+| CSS: 3 levels of selector depth as max                                                             |                                                                                     |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/concepts/basic-conventions.md:19]: Rather: "CSS Selectors" on the left side, "3 levels of selector depth as max" on the right side

File: products/editions/enterprise-edition/concepts/basic-conventions.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+| `{block name="b2b_*"}{/block}` empty blocks are in one line                                        |                                                                                     |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/concepts/basic-conventions.md:20]: Outdated syntax. With twig, this should be 

`{% block b2b_* %}{% endblock %}`

File: products/editions/enterprise-edition/concepts/basic-conventions.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+| JavaScript                                                                                         | File name looks like `*.plugin.ts`                                                  |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/concepts/basic-conventions.md:21]: Rather:
"JavaScript | The B2B Suite is written in TypeScript"
"Storefront Plugins | File names end with *.plugin.ts"
"Interfaces | File names start with `I`, e.g. `IAjaxPanelEvent.ts`

File: products/editions/enterprise-edition/concepts/method-structure.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+```
+{% endcode %}
+
+### Public functions in framework
+
+Public functions are made to be compatible and not be changed until major version changes.
+
+### TypeScript functions
+
+TypeScript functions always have access modifiers and are completely typed with their arguments and return types.
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/concepts/method-structure.md:47]: You could add, that the same deprecation rules apply here.

File: SUMMARY.md (hunk 1 above)
Comment (Dominik28111) [reply SUMMARY.md:43]: ah my fault 😅

**Comment by Dominik28111** [comment]
create b2b-suite folder and put content into it.

**Review by Dominik28111 — COMMENTED** [review]
create b2b-suite folder and put content into it.

File: products/editions/enterprise-edition/b2b-suite/concepts/basic-conventions.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+| Tests                                                                                              | All test methods are in snake case                                                  |
+| All test methods start with `test_`                                                                |                                                                                     |
+| Templates                                                                                          | All new layout modules are wrapped in `b2b--*` class containers                     |
+| Modules reuse the template style of shopware                                                       |                                                                                     |
+| CSS Selectors                                                                                      | 3 levels of selector depth as max                                                   |
+| {% raw %}`{% block b2b_* %}{% endblock %}`{% endraw %} empty blocks are in one line                |                                                                                     |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/b2b-suite/concepts/basic-conventions.md:20]: Move the description to the right-hand side column

File: products/editions/enterprise-edition/b2b-suite/concepts/basic-conventions.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+| JavaScript                                                                                         | The B2B Suite is written in TypeScript                                              |
+| Storefront Plugins                                                                                 | File names end with *.plugin.ts                                                     |
+| Interfaces                                                                                         | File names start with `I`, e.g. `IAjaxPanelEvent.ts`                                |
+| Snippets                                                                                           |                                                                                     |
+| The root snippet key is `b2b`                                                                      |                                                                                     |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/b2b-suite/concepts/basic-conventions.md:25]: Move the description to the right-hand side column

**Review by bschulzebaek — CHANGES_REQUESTED** [review]

File: products/editions/enterprise-edition/b2b-suite/concepts/system-architecture.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+A single component with all layers and the maximum of allowed dependencies looks like this:
+
+![image](../../../../.gitbook/assets/b2b-architecture-component.svg)
+
+The responsibilities from bottom to top:
+
+| Layer       | Description                                                                                                                                                                                                      |
+|-------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Shop-Bridge | Bridges the broad Shopware interfaces to the specific framework requirements <ul><li>Implements interfaces provided by the framework</li><li>Subscribes to shopware events and calls framework services</li><ul> |
```
Comment (bschulzebaek) [inline products/editions/enterprise-edition/b2b-suite/concepts/system-architecture.md:15]: Bildschirmfoto 2022-04-26 um 11 28 38

The list is broken.

**Review by bschulzebaek — CHANGES_REQUESTED** [review]

File: products/editions/enterprise-edition/b2b-suite/concepts/method-structure.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+[...]
+
+class GridHelper
+{    
+    [...]
+    
+    /**
+     * @internal
+     */
+    protected function extractLimitAndOffset(Request $request, SearchStruct $struct)
```
Comment (MartinKrzykawski) [inline products/editions/enterprise-edition/b2b-suite/concepts/method-structure.md:30]: Please add void as return type

**Review by bschulzebaek — APPROVED** [review]

**Review by MartinKrzykawski — APPROVED** [review]

**Pull request merged**